server/src/future.rs

```diff
@@ -30,19 +30,27 @@ use jsonrpsee_core::Error;
 use std::sync::Arc;
 use tokio::sync::{watch, OwnedSemaphorePermit, Semaphore, TryAcquireError};
 
+/// Create channel to determine whether
+/// the server shall continue to run or not.
+pub fn stop_channel() -> (StopHandle, ServerHandle) {
+	let (tx, rx) = tokio::sync::watch::channel(());
+	(StopHandle::new(rx), ServerHandle::new(tx))
+}
+
 /// Represent a stop handle which is a wrapper over a `multi-consumer receiver`
 /// and cloning [`StopHandle`] will get a separate instance of the underlying receiver.
 #[derive(Debug, Clone)]
-pub(crate) struct StopHandle(watch::Receiver<()>);
+pub struct StopHandle(watch::Receiver<()>);
 
 impl StopHandle {
-	pub(crate) fn new(rx: watch::Receiver<()>) -> Self {
+	/// Create a new stop handle.
+	pub fn new(rx: watch::Receiver<()>) -> Self {
```

Comment:
I wonder whether this can be `pub(crate)` again now or whethery ou'd prefer it was pub?